test/e2e/cert-manager_test.go

```diff
@@ -0,0 +1,72 @@
+package e2e
+
+import (
+	"context"
+	"testing"
+
+	"github.com/stretchr/testify/assert"
+	"k8s.io/apimachinery/pkg/api/errors"
+	"sigs.k8s.io/e2e-framework/pkg/envconf"
+	"sigs.k8s.io/e2e-framework/pkg/features"
+
+	certmanager "go.etcd.io/etcd-operator/pkg/certificate/cert-manager"
+	interfaces "go.etcd.io/etcd-operator/pkg/certificate/interfaces"
+)
+
+func TestCertManagerCM(t *testing.T) {
+	feature := features.New("Cert-Manager Certificate").WithLabel("app", "cert-manager")
+
+	feature.Setup(
```

Comment:
@abdurrehman107 @ahrtr I was wondering if we can break [TestMain()](https://github.com/etcd-io/etcd-operator/blob/main/test/e2e/e2e_suite_test.go#L44-L205) into separate funcs like `CreateKindCluster`, `InstallPrometheus`, `InstallCertmanager`, `DeployOperator`?


Separate funcs will allow us to import these into any future e2e tests we would want to write, something like this: https://github.com/kubernetes-sigs/apiserver-network-proxy/blob/master/e2e/main_test.go#L174

WDYT?

Reply:
> @abdurrehman107 @ahrtr I was wondering if we can break [TestMain()](https://github.com/etcd-io/etcd-operator/blob/main/test/e2e/e2e_suite_test.go#L44-L205) into separate funcs like `CreateKindCluster`, `InstallPrometheus`, `InstallCertmanager`, `DeployOperator`?

Is it blocking your task? If no, please focus on your current task for now.

Reply:
Its not a blocker now, it can be done in the future